src/overlays/actors/ovl_Bg_Jya_Cobra/z_bg_jya_cobra.c

```diff
@@ -26,41 +52,628 @@ const ActorInit Bg_Jya_Cobra_InitVars = {
+    sp44.y = this->dyna.actor.shape.rot.y;
+    sp44.z = this->dyna.actor.shape.rot.z;
+    func_800D1694(this->unk_180.x, this->unk_180.y, this->unk_180.z, &sp44);
+
+    Matrix_Scale(0.1f, 0.1f, this->unk_190, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 939),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, (s32)(this->unk_18C * 140.0f));
+    gSPDisplayList(POLY_XLU_DISP++, D_060117D0);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 947);
+}
+
+void func_80896EE4(BgJyaCobra* this, GlobalContext* globalCtx) {
+    u32 temp;
+    s16 sp72;
+    Vec3f sp64;
+    Vec3s* phi_a3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/BgJyaCobra_Init.s")
+    sp72 = this->dyna.actor.params & 3;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/BgJyaCobra_Destroy.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 966);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896918.s")
+    func_80094044(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896950.s")
+    if (sp72 == 0) {
+        sp64.x = this->dyna.actor.posRot.pos.x - 50.0f;
+        sp64.y = this->dyna.actor.posRot.pos.y;
+        sp64.z = this->dyna.actor.posRot.pos.z;
+        phi_a3 = &D_80897538;
+    } else {
+        phi_a3 = &this->dyna.actor.shape.rot;
+        if (sp72 == 2) {
+            sp64.x = this->dyna.actor.posRot.pos.x + 70.0f;
+            sp64.y = this->dyna.actor.posRot.pos.y;
+            sp64.z = this->dyna.actor.posRot.pos.z;
+            phi_a3 = &D_80897540;
+        } else {
+            Math_Vec3f_Copy(&sp64, &this->dyna.actor.posRot.pos);
+        }
+    }
+    func_800D1694(sp64.x, sp64.y, sp64.z, phi_a3);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_808969F8.s")
+    Matrix_Scale(D_80897548[sp72].x, D_80897548[sp72].y, D_80897548[sp72].z, MTXMODE_APPLY);
+    Matrix_Translate(0.0f, 0.0f, 40.0f, MTXMODE_APPLY);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Cobra/func_80896ABC.s")
+    gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 0, 0, 0, 120);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_jya_cobra.c", 994),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gDPSetTextureImage(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 1, ALIGN16((s32)(&this->unk_194)));
+    gDPSetTile(POLY_XLU_DISP++, G_IM_FMT_I, G_IM_SIZ_16b, 0, 0x0000, G_TX_LOADTILE, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+               G_TX_NOMASK, G_TX_NOLOD, G_TX_NOMIRROR | G_TX_CLAMP, G_TX_NOMASK, G_TX_NOLOD);
+    gDPLoadSync(POLY_XLU_DISP++);
+    temp = sp72;
+    gDPLoadBlock(POLY_XLU_DISP++, G_TX_LOADTILE, 0, 0, 2047, 256);
+    if (!temp) {}
```

Comment:
```suggestion
    if (1) {}
```
Try doing this instead, make sure it matches first though.